src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -521,12 +522,36 @@ void AutomatedRssDownloader::handleFeedCheckStateChange(QListWidgetItem *feed_it
 void AutomatedRssDownloader::updateMatchingArticles()
 {
     ui->treeMatchingArticles->clear();
+    saveEditedRule();
+    m_updateMatchingArticlesDeferrals.ref();
+    QTimer::singleShot(1000, this, SLOT(deferredUpdateMatchingArticles()));
+}
+
+void AutomatedRssDownloader::deferredUpdateMatchingArticles()
+{
+    // Wait until there's been no updates at all for 1 second
+    if (m_updateMatchingArticlesDeferrals.deref())
+        return;
+
+    ui->treeMatchingArticles->clear();
+
     Rss::ManagerPtr manager = m_manager.toStrongRef();
-    if (!manager)
+    if (!manager) {
+        ui->treeMatchingLoading->setPixmap(QPixmap());
         return;
+    }
+
+    if (( ui->treeMatchingLoading->pixmap() == 0) || ui->treeMatchingLoading->pixmap()->isNull()) {
+        // Set the loading icon and allow it to be displayed
+        ui->treeMatchingLoading->setPixmap(QIcon(":/icons/loading.png").pixmap(16, 16));
+        m_updateMatchingArticlesDeferrals.ref();
+        // Need to have a 1ms delay here or we don't get the screen update - not sure why
```

Comment:
That's because UI updates happen after control has returned to the mainloop.
So instead of the timer+return I think you can safely do:
```c++
QCoreApplication::instance()->processEvents(QEventLoop::ExcludeUserInputEvents | QEventLoop::ExcludeSocketNotifiers);
```